packages/experiments/src/components/Toggle/Toggle.types.ts

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { IComponent, IStyleableComponentProps } from '../../Foundation';
+import { IKeytipProps } from 'office-ui-fabric-react/lib/Keytip';
+import { IStyle } from 'office-ui-fabric-react/lib/Styling';
+import { IComponentAs /*, IRefObject*/ } from '../../Utilities';
+
+export type IToggleComponent = IComponent<IToggleProps, IToggleViewProps, IToggleStyles>;
+
+/**
+ * Toggle component props.
+ */
+export interface IToggleProps extends IStyleableComponentProps<IToggleViewProps, IToggleStyles> {
+  /**
+   * Render the root element as another type.
+   */
+  as?: IComponentAs<React.HTMLAttributes<HTMLElement>>;
+
+  /**
+   * Optional callback to access the IToggleComponent interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  // componentRef?: IRefObject<IToggleComponent>;
+
+  /**
+   * A label for the toggle.
+   */
+  label?: string;
+
+  /**
+   * Text to display when toggle is ON.
+   */
+  onText?: string;
+
+  /**
+   * Text to display when toggle is OFF.
+   */
+  offText?: string;
+
+  /**
+   * Text for screen-reader to announce as the name of the toggle.
+   */
+  ariaLabel?: string;
+
+  /**
+   * Initial state of the toggle. If you want the toggle to maintain its own state, use this. Otherwise refer to `checked`.
+   * @default false
+   */
+  defaultChecked?: boolean;
+
+  /**
+   * Checked state of the toggle. If you are maintaining state yourself, use this property. Otherwise refer to `defaultChecked`.
+   * @default defaultChecked
+   */
+  checked?: boolean;
+
+  /**
+   * Optional disabled flag.
+   * @default false
+   */
+  disabled?: boolean;
+
+  /**
+   * Callback issued when the value changes.
+   */
+  onChange?: (event: React.MouseEvent<HTMLElement>, checked?: boolean) => void;
+
+  /**
+   * Optional keytip for this toggle
+   */
+  keytipProps?: IKeytipProps;
+}
+
+export type IToggleViewProps = Pick<
+  IToggleProps,
+  'as' /*| 'componentRef'*/ | 'label' | 'ariaLabel' | 'disabled' | 'onChange' | 'keytipProps'
```

Comment:
Actually as I think of this view shouldn't ever need `componentRef` prop as the state component will have the interface implementation (and will therefore be the actual component reference.)

Reply:
I removed the prop from the ViewProps list 👍.